pkg/ibmcsidriver/controller_helper.go

```diff
@@ -279,6 +309,13 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 		volume.Az = zones[utils.NodeZoneLabel]
 	}
 
+	// Validation for primaryIPAddress and primaryIPID
+	if volume.VPCVolume.PrimaryIP != nil && len(volume.VPCVolume.PrimaryIP.ID) > 0 && len(volume.VPCVolume.PrimaryIP.Address) > 0 {
+		err = fmt.Errorf("invalid option either provide primaryIPID or primaryIPAddress: '%v'", err)
```

Comment:
Sameer.. I assume this is to cover
`Use Case 7: Negative security_group Mode (ENI) with custom security group and user provided primary IPAddress and primaryIPID`

Could you confirm my understanding is correct

Reply:
Yes

Reply:
Ideally the reason I put the check in Driver is if we don't then it will go in VPC and fail with same reason so to avoid unwanted calls